sympy/sets/sets.py

```diff
@@ -165,7 +297,10 @@ def _union(self, other):
 
         Used within the :class:`Union` class
         """
-        return None
+        try:
+            return self._union_simplify(other)
+        except:
```

Comment:
Please avoid bare except clauses. You should always catch the exception you are looking for, so that unrelated exceptions can bubble up. A bare `except` will catch everything, even things like `RuntimeError` and `KeyboardInterrupt`.